src/cpp/src/visual_language/vision_encoder.hpp

```diff
@@ -76,6 +77,25 @@ class VisionEncoder {
         ov::Core core=ov::Core{}
     );
 
+    /// @brief Construct the encoder from models map.
+    /// @param model Model IR as string (openvino_vision_embeddings_model.xml)
+    /// @param weights Model weights as tensor (openvino_vision_embeddings_model.bin)
+    /// @param config_dir_path A path to directory containing preprocessor_config.json.
+    /// @param model_type A type of VLM model.
+    /// @param device A device to compile the encoder for.
+    /// @param device_config A config to be passed to
+    /// ov::Core::compile_model().
+    /// @param core ov::Core to be used to compile the model.
+    explicit VisionEncoder(
+        const std::string& model,
+        const ov::Tensor& weights,
+        const std::filesystem::path& config_dir_path,
+        const VLMModelType model_type,
+        const std::string& device="CPU",
+        const ov::AnyMap device_config={},
+        ov::Core core=ov::Core{}
```

Comment:
@Wovchena do we need `core` ? I suppose we are all using singleton_core now.

Reply:
Removed `core` parameter from vision encoder constructor, used `singleton_core()` in instead